packages/xstate-graph/src/graph.ts

```diff
@@ -435,19 +451,31 @@ export function performDepthFirstTraversal<TState, TEvent extends EventObject>(
   const adj: AdjacencyMap<TState, TEvent> = {};
 
   let iterations = 0;
-  const queue: Array<[TState, TEvent | null]> = [[initialState, null]];
+  const queue: Array<
+    [
```

Comment:
This named tuple is awkward; should probably be an object. Will be a future refactor.